src/openslide-decode-tiff.c

```diff
@@ -683,3 +683,57 @@ void _openslide_tiffcache_destroy(struct _openslide_tiffcache *tc) {
   g_free(tc->filename);
   g_free(tc);
 }
+
+bool _openslide_tiff_icc_profile_size(openslide_t *osr G_GNUC_UNUSED,
+                                      struct _openslide_tiff_level *tiffl,
+                                      TIFF *tiff,
+                                      int64_t *len,
+                                      GError **err) {
+  // set directory
+  if (!_openslide_tiff_set_dir(tiff, tiffl->dir, err)) {
+    return false;
+  }
+
+  // get profile size
+  guint32 data_len;
+  void *data;
+  if (TIFFGetField(tiff, TIFFTAG_ICCPROFILE, &data_len, &data)) {
+    *len = data_len;
+  } else {
+    // 0 means no profile available
+    *len = 0;
+  }
+
+  return true;
+}
+
+bool _openslide_tiff_icc_profile_read(openslide_t *osr G_GNUC_UNUSED,
```

Comment:
Let's move this earlier, next to the other functions that operate on TIFFs.

Why are we taking an `osr` if we don't use it?

Reply:
I moved it. We'll need osr to validate the length, I think.